backend/src/controllers/user.ts

```diff
@@ -0,0 +1,66 @@
+import { Request, Response } from "express";
+import { validationResult } from "express-validator";
+import admin from "firebase-admin";
+
+import { UserError } from "../errors";
+import { errorHandler } from "../errors/handler";
+import { firebaseAuth } from "../firebase/firebase_config";
+import User, { UserDocument } from "../models/User";
+
+// Define the type for req.body
+type CreateUserRequestBody = {
+  name: string;
+  accountType: "admin" | "team";
+  approvalStatus: boolean;
+  email: string;
+  password: string;
+};
+
+export const createUser = async (
+  req: Request<Record<string, never>, Record<string, never>, CreateUserRequestBody>,
+  res: Response,
+) => {
+  try {
+    // Check for validation errors
+    const errors = validationResult(req);
+    // if (!errors.isEmpty()) {
+    //   return res.status(400).json({ errors: errors.array() });
+    // }
+    if (!errors.isEmpty()) {
+      let errStr = "";
+      for (const err of errors.array()) {
+        errStr += err.msg + " ";
+      }
+      errStr = errStr.trim();
+      throw new UserError(0, 400, errStr);
+    }
+
+    const { name, accountType, approvalStatus, email, password } = req.body;
```

Comment:
Let's not make approval status something that is passed in as a parameter in the req.body. This should be set to false by default in the controller itself. That way no bad actors can call the route and create an account with approval status true, and gain access to the database.